packages/core/getAggregatedAvailability/index.ts

```diff
@@ -4,6 +4,22 @@ import { SchedulingType } from "@calcom/prisma/enums";
 
 import { mergeOverlappingDateRanges } from "./date-range-utils/mergeOverlappingDateRanges";
 
+function uniqueAndSortedDateRanges(ranges: DateRange[]): DateRange[] {
+  const seen = new Set<number>();
+
+  return ranges
+    .sort((a, b) => {
+      const startDiff = a.start.valueOf() - b.start.valueOf();
+      return startDiff !== 0 ? startDiff : a.end.valueOf() - b.end.valueOf();
+    })
+    .filter((range) => {
+      const key = range.start.valueOf() * 1e12 + range.end.valueOf();
```

Comment:
The highlighted line const key = range.start.valueOf() * 1e12 + range.end.valueOf(); is part of the uniqueAndSortedDateRanges function. This line generates a unique key for each date range by combining the start and end timestamps. Multiplying the start timestamp by 1e12 ensures that the start and end values are distinct and non-overlapping when combined, allowing the function to accurately identify and filter out duplicate date ranges.

Reply:
This is not working correctly for me. 

I have the following data: 

Date range 1: 
start: 2025-01-27T14:00:00Z
end: 2025-01-27T04:30:00-05:00

Date range 2: 
start: 2025-01-27T14:00:00Z
end: 2025-01-27T14:45:00Z

Even though the end time is different, Date Range 2 is considered a duplicate and not returned